src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2642,6 +2642,13 @@ def transform(dataset):
             await self.middleware.call(
                 'notifier.change_dataset_share_type', data['name'], data.get('share_type', 'UNIX').lower()
             )
+            if data.get('share_type', 'UNIX') == 'WINDOWS':
+                dataset = await self.middleware.call('zfs.dataset.query', [('name', '=', data['name'])])
+                setacl_job = await self.middleware.call('filesystem.setacl', dataset[0]['mountpoint'], [
+                    {"tag": "owner@", "id": None, "type": "ALLOW", "perms": {"BASIC": "FULL_CONTROL"}, "flags": {"BASIC": "INHERIT"}},
+                    {"tag": "group@", "id": None, "type": "ALLOW", "perms": {"BASIC": "FULL_CONTROL"}, "flags": {"BASIC": "INHERIT"}}
+                ])
+                await setacl_job.wait()
```

Comment:
We should probably either log or raise if setacl fails?
if setacl_job.error:
    # something